### PR TITLE
Prometheus: Add exemplar modal "locking" for time series

### LINKED ISSUE #64061: Prometheus: Time series and histogram exemplar UX differences
**What happened**:
Exemplars are not clickable in time series panel, but are for histograms, makes dashboards with mixed panel types confusing and frustrating for users.

**What you expected to happen**:
Exemplar UX should be consistent across different panel options.

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**
Allowing users to click on the Exemplar in time series panel to lock the modal in place, until the user closes the modal, to more closely match the UX of the heatmap exemplars.

**Why do we need this feature?**
To unify UX between time-series and histogram exemplars. Clicking on an exemplar in the canvas implementation used in the histogram will also lock the modal in place and prevent the hover action from opening other exemplars in the same panel, which helps out immensely when dealing with large modals and/or densely clustered exemplars.

**Who is this feature for?**
Prometheus exemplar users.

**Which issue(s) does this PR fix?**:
Fixes https://github.com/grafana/grafana/issues/64061

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/panel/timeseries/plugins/ExemplarMarker.tsx [hunk 1]
```diff
@@ -122,8 +146,31 @@ export const ExemplarMarker: React.FC<ExemplarMarkerProps> = ({
         {...attributes.popper}
       >
         <div className={styles.wrapper}>
+          {isLocked && (
```
Comment (gtk-grafana) [inline public/app/plugins/panel/timeseries/plugins/ExemplarMarker.tsx:149]: can we make this a component? I think it's a direct copy/paste from the histogram ya lazy punk

File: public/app/plugins/panel/timeseries/TimeSeriesPanel.tsx [hunk 2]
```diff
@@ -133,13 +142,21 @@ export const TimeSeriesPanel = ({
...
             )}
             {data.annotations && (
-              <ExemplarsPlugin
-                visibleSeries={getVisibleLabels(config, frames)}
-                config={config}
-                exemplars={data.annotations}
-                timeZone={timeZone}
-                getFieldLinks={getFieldLinks}
-              />
+              <ClickOutsideWrapper
```
Comment (gtk-grafana) [inline public/app/plugins/panel/timeseries/TimeSeriesPanel.tsx:145]: I'm not sure you should include this or not with this PR, but it's a nicer UX

Comment (gtk-grafana) [reply public/app/plugins/panel/timeseries/TimeSeriesPanel.tsx:145]: No I lied pull it

**Comment by leeoniya** [comment]
>  also we want to investigate if this is something that can be fixed for all portal windows.

have you seen https://github.com/grafana/grafana/pull/63593?

**Comment by gtk-grafana** [comment]
> > also we want to investigate if this is something that can be fixed for all portal windows.
> 
> have you seen #63593?

No, that's awesome tho, after that gets merged we should use that for the new unified exemplars?

**Comment by gtk-grafana** [comment]
@leeoniya That base ticket was merged in now so this should be easier to review

**Review by bohandley — APPROVED** [review]
Clicked exemplar and the modal stayed, looks nice! LGTM

https://user-images.githubusercontent.com/25674746/231777872-c2f14c92-ae7d-478d-9bb0-476cfef54f93.mov

**Pull request merged**